File: pset1/cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int get_valid_input(string prompt);

int main(void)
{
    // Return value in coins
    int coins = get_valid_input("Change owed: ");
    
    // Start performing checks
    int amount = 0;
    while (coins >= 25)
    {
        coins = coins - 25;
        amount++;
    }
    while (coins >= 10)
    {
        coins = coins - 10;
        amount++;
    }
    while (coins >= 5)
    {
        coins = coins - 5;
        amount++;
    }
    while (coins >= 1)
    {
        coins = coins - 1;
        amount++;
    }
    
    // Print coin amount
    printf("%d\n", amount);
}

// Prompt user for float input and check for validity
// Then convert to cents and integer value type
int get_valid_input(string prompt)
{
    float dollars;
    int coins;
    do
    {
        dollars = get_float("%s", prompt);
        coins = round(dollars * 100);
    }
    while (coins <= 0);
    return coins;
}

File: pset6/cash.py
from cs50 import get_float


def main():
    while True:
        # getting input
        change = get_float('Change owed: ')

        # converting to cents
        change = int(change * 100)

        # check for correct usage
        if change > 0:
            break

    # we start with zero coins/change
    coins = 0

    # start performing checks from largest to smallest coin
    while change >= 25:
        change = change - 25
        coins += 1

    while change >= 10:
        change = change - 10
        coins += 1

    while change >= 5:
        change = change - 5
        coins += 1

    while change >= 1:
        change = change - 1
        coins += 1

    # print output
    print(coins)


if __name__ == "__main__":
    main()